File: Java/Medicine.java


import java.util.ArrayList;
import java.util.Collection;

public class Medicine {
    
    private Collection<Prescription> prescriptions = new ArrayList<Prescription>();
    
    private final String name;

    public Medicine(String name) {
        this.name = name;
    }
    
    public void addPrescription(Prescription prescription) {
        this.prescriptions.add(prescription);
    }
}

File: JavaScript/Medicine.js
function Medicine(name) {
    "use strict";

    this.prescriptions = [];
    this.name = name;

}

Medicine.prototype.getName = function () {
    return this.name;
}

Medicine.prototype.addPrescription = function (prescription) {
    this.prescriptions.push(prescription);
}

// var med = Object.create(Medicine.prototype);